src/python/pants/backend/python/rules/run_setup_py.py

```diff
@@ -469,25 +472,25 @@ async def determine_setup_kwargs(
     exported_target: ExportedTarget, union_membership: UnionMembership
 ) -> SetupKwargs:
     target = exported_target.target
-    plugin_requests = union_membership.get(SetupKwargsPluginRequest)  # type: ignore[misc]
+    plugin_requests = union_membership.get(SetupKwargsRequest)  # type: ignore[misc]
```

Comment:
I'm wondering if we can centralize the logic of "find 0 or 1 union members applicable to a target (or a param more generally), or error" in the UnionMembership? Can be in a followup.

Reply:
Possibly, but I'm not sure how generalizable the specific code is, even though the general idiom is used frequently. 

For example, the error messages differ for each implementation so that they can be specific to the application, e.g. `TargetsToValidFieldSets` is sometimes okay if there are zero matches, but it sometimes errors in that case. Here, if there are no matches, we use the default case.